core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -6466,6 +6466,32 @@ private void checkLiteral2(String expression, String expected) {
         .withBigQuery().ok(expected);
   }
 
+  @Test void testBigQueryArrayElementAccessing() {
+    final String arrayCtorSql = ""
+        + "SELECT ARRAY[1, 2, 3][1]";
+    final String expectedArrayCtorSql = ""
+        + "SELECT ARRAY[1, 2, 3][ORDINAL(1)]";
+    sql(arrayCtorSql)
+        .withBigQuery().ok(expectedArrayCtorSql);
+
+    // TODO: Got error from
+    //  `SqlImplementor`: Need to implement org.apache.calcite.rel.core.Collect
```

Comment:
I tried to add test for `ARRAY_QUERY` but got this error. Should we create another PR for it?